### ISSUE TITLE
"WIP" app did not prevent merge

### ISSUE THREAD

**Issue by clxmstaab** [chdsbd/kodiak#168]
today we opened a pull request with the title-prefix "WIP".

the WIP  check detected the use of this keyword and marked the check as "yellow".
after the travis build succeeded kodiak immediately merged the PR.

![image](https://user-images.githubusercontent.com/47448731/64102193-ba2bb000-cd6f-11e9-9971-c3edd967a165.png)

we expected that kodiak should wait with merging the PR until we remove the "WIP" title from the pull request

our .kodiak.toml looks like
```
# .kodiak.toml
version = 1

[merge]
method = "squash"
# once a PR is merged into master, delete the branch
delete_branch_on_merge = true
dont_wait_on_status_checks = ["WIP"] # handle github.com/apps/wip
# label to use to enable Kodiak to merge a PR
automerge_label = "automerge" # default: "automerge"
# require that the automerge label be set for Kodiak to merge a PR. if you
# disable this Kodiak will immediately attempt to merge every PR you create
require_automerge_label = true

[merge.message]
title = "pull_request_title"
body = "github_default"
include_pr_number = true
strip_html_comments = true # remove html comments to auto remove PR templates
```

**Comment by clxmstaab** [comment]
hmm I get the feeling that kodiak does not wait for the WIP github marketplace app and in my initial tests in which things worked, it worked because of the `blacklist_title_regex` setting.

in my recent test I used the prefix `WIP` but kodiak expects `WIP:` (and the README contains a typo): https://github.com/chdsbd/kodiak/pull/169

**Comment by chdsbd** [comment]
I think the problem is that WIP isn’t a required check on your repo. If you add it to the branch protection settings kodiak will wait for it. 

blacklist_title_regex will prevent a PR from being merged if it matches the regex, but I don’t think it matches the WIP app behavior 100%.

You must use dont_wait_on_status_checks with the WIP app to prevent kodiak from getting stuck waiting for the WIP check to pass.

**Comment by clxmstaab** [comment]
thanks for your detailed response.

> I think the problem is that WIP isn’t a required check on your repo. If you add it to the branch protection settings kodiak will wait for it.

you are right.. the branch protection was not configured properly.

> blacklist_title_regex will prevent a PR from being merged if it matches the regex, but I don’t think it matches the WIP app behavior 100%.


I guess the blacklist_title_regex is then good enough for us. we will use it instead of adding another external service to our builds.
